akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/ibex35/Ibex35HistoricManager.scala

```diff
@@ -0,0 +1,120 @@
+package com.tapiporla.microservices.retrievers.indices.ibex35
+
+import akka.actor.{Props, Stash}
+import com.tapiporla.microservices.retrievers.common.{ElasticDAO, Retriever}
+import com.tapiporla.microservices.retrievers.common.Retriever.UpdateHistoricData
+import com.tapiporla.microservices.retrievers.common.ElasticDAO._
+import com.tapiporla.microservices.retrievers.indices.ibex35.Ibex35HistoricManager.{InitIbex35Coordinator, ReadyToStart, UpdateComplete}
+import com.tapiporla.microservices.retrievers.indices.ibex35.dao.Ibex35ScrapyDAO.{CantRetrieveDataFromIbex35Crawler, IbexDataRetrieved, RetrieveAllIbexData, RetrieveIbexDataFrom}
+import com.tapiporla.microservices.retrievers.indices.ibex35.dao.{Ibex35ESDAO, Ibex35ScrapyDAO}
+import com.tapiporla.microservices.retrievers.indices.ibex35.model.Ibex35Historic
+import org.elasticsearch.search.sort.SortOrder
+import org.joda.time.DateTime
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+
+
+object Ibex35HistoricManager {
+  object InitIbex35Coordinator
+  case class ReadyToStart(lastUpdated: Option[DateTime])
+  object UpdateComplete
+}
+
+/**
+  * Actor in charge of the Ibex35 data retrieved from ScrapyRT
+  * and persisted in ElasticSearch
+  */
+class Ibex35HistoricManager extends Retriever with Stash {
+
+  val esDAO =
+    context.actorOf(Props[Ibex35ESDAO], name = "Ibex35ESDAO_Historic")
+
+  val scrapyDAO =
+    context.actorOf(Props[Ibex35ScrapyDAO], name = "Ibex35ScrapyDAO_Historic")
+
+  self ! InitIbex35Coordinator
+
+  override def receive = initial
+
+  def initial: Receive = {
+
+    case InitIbex35Coordinator =>
+      import com.sksamuel.elastic4s.ElasticDsl._
+      log.info("Recovering initial status")
+      esDAO ! RetrieveAllFromIndexSorted(
+        Ibex35ESDAO.index,
+        Ibex35ESDAO.Historic.typeName,
+        None,
+        Some(fieldSort(Ibex35ESDAO.date).order(SortOrder.DESC)),
+        Some(1)
+      )
+
+    case DataRetrieved(index, typeName, data, _) =>
+      if(data.isEmpty) {
```

Comment:
you could use 
```scala
data.headOption.fold{
  /*case where is empty*/
}{ head => 
  /*case where it's not*/ 
}
```

Reply:
Ooops, didn't read properly. Forget about this

Reply:
Done. And thank you very much :)